### PR TITLE
[Inference API] Add Docs for Amazon Bedrock Support for the Inference API

### PR DESCRIPTION
**Pull request by markjhoy**
Add docs in support of Amazon Bedrock support in the Inference API: https://github.com/elastic/elasticsearch/pull/110248

### CODE REVIEW

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+Note that the model ID must be available for the provider chosen, and your IAM user must have access to the model.
+
+`region`:::
+(Required, string)
+The region that your model or ARN is deployed in.
+The list of available regions per model can be found in the https://docs.aws.amazon.com/bedrock/latest/userguide/models-regions.html[Model support by AWS region] documentation.
+
+`rate_limit`:::
+(Optional, object)
+By default, the `azureaistudio` service sets the number of requests allowed per minute to `240`.
```
Comment (timgrein) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:88]:
```suggestion
By default, the `amazonbedrock` service sets the number of requests allowed per minute to `240`.
```

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+This helps to minimize the number of rate limit errors returned from Azure AI Studio.
```
Comment (timgrein) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:89]:
```suggestion
This helps to minimize the number of rate limit errors returned from Amazon Bedrock.
```

**Review by timgrein — COMMENTED** [review]
Just highlighting some Azure AI Studio references (Sorry, I just saw that this is draft, but leaving the comments here, so we don't forget it :) )

File: docs/reference/inference/service-amazon-bedrock.asciidoc (hunk 2 above)
Comment (markjhoy) [reply docs/reference/inference/service-amazon-bedrock.asciidoc:89]: Argh - great catches ;) That's what I get for copy / pasting

**Comment by markjhoy** [comment]
@elasticmachine run docs build

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+(Optional, float)
+Only available for `anthropic`, `cohere`, and `mistral` providers.
+A number in the range of 0.0 to 1.0 that is an alternative value to temperature or top_p that causes the model to consider the results of the tokens with nucleus sampling probability.
+Should not be used if `temperature` or `top_p` is specified.
+
+=====
++
+.`task_settings` for the `text_embedding` task type
+[%collapsible%closed]
+=====
```
Comment (leemthompo) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:129]: @markjhoy I think this unclosed `====` block might be breaking your build :)

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+(Optional, float)
+A number in the range of 0.0 to 1.0 that specifies the sampling temperature to use that controls the apparent creativity of generated completions.
+Should not be used if `top_p` or `top_k` is specified.
+
+`top_p`:::
+(Optional, float)
+A number in the range of 0.0 to 1.0 that is an alternative value to temperature that causes the model to consider the results of the tokens with nucleus sampling probability.
+Should not be used if `temperature` or `top_k` is specified.
+
+`top_p`:::
```
Comment (leemthompo) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:119]:
```suggestion
`top_k`:::
```

Assuming the first top_p is the correct one 😉

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+(Optional, object)
+include::inference-shared.asciidoc[tag=task-settings]
++
+.`task_settings` for the `completion` task type
+[%collapsible%closed]
+=====
+
+`max_new_tokens`:::
+(Optional, integer)
+Provides a hint for the maximum number of output tokens to be generated.
```
Comment (leemthompo) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:106]:
```suggestion
Sets a maximum number for the output tokens to be generated.
```

Not sure what "hint" means here, rewording tries to clarify

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+Defaults to 64.
+
+`temperature`:::
+(Optional, float)
+A number in the range of 0.0 to 1.0 that specifies the sampling temperature to use that controls the apparent creativity of generated completions.
```
Comment (leemthompo) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:111]:
```suggestion
A number between 0.0 and 1.0 that controls the apparent creativity of the results. At temperature 0.0 the model is most deterministic, at temperature 1.0 most random.
```

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+Should not be used if `top_p` or `top_k` is specified.
+
+`top_p`:::
+(Optional, float)
+A number in the range of 0.0 to 1.0 that is an alternative value to temperature that causes the model to consider the results of the tokens with nucleus sampling probability.
```
Comment (leemthompo) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:116]:
```suggestion
Alternative to `temperature`. A number in the range of 0.0 to 1.0, to eliminate low-probability tokens. Top-p uses nucleus sampling to select top tokens whose sum of likelihoods does not exceed a certain value, ensuring both variety and coherence.
```

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+Should not be used if `temperature` or `top_k` is specified.
```
Comment (leemthompo) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:117]: Reading around it looks like top-p and top-k can be used in combination?

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+`top_p`:::
+(Optional, float)
+Only available for `anthropic`, `cohere`, and `mistral` providers.
+A number in the range of 0.0 to 1.0 that is an alternative value to temperature or top_p that causes the model to consider the results of the tokens with nucleus sampling probability.
```
Comment (leemthompo) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:122]:
```suggestion
Alternative to `temperature`. Limits samples to the top-K most likely words, balancing coherence and variability.
A number in the range of 0.0 to 1.0.
```

File: docs/reference/inference/service-amazon-bedrock.asciidoc [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+
+There are no `task_settings` available for the `text_embedding` task type.
+
+[discrete]
+[[inference-example-amazonbedrock]]
+==== Amazon Bedrock service example
+
+The following example shows how to create an {infer} endpoint called `amazon_bedrock_embeddings` to perform a `text_embedding` task type.
+
+The list of chat completion and embeddings models that you can choose from should be a https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html[Amazon Bedrock base model] you have access to.
```
Comment (leemthompo) [inline docs/reference/inference/service-amazon-bedrock.asciidoc:139]:
```suggestion
Choose chat completion and embeddings models you can access from the https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html[Amazon Bedrock base models].
```
nit: keep sentence short

**Review by leemthompo — COMMENTED** [review]
This is looking good. Found a few minor errors and suggested some rephrasings. Also hopefully identified the formatting issues that's failing docs build. Once these updates are made and we can preview the formatting for tabs, will be ready for final review!

File: docs/reference/inference/service-amazon-bedrock.asciidoc (hunk 3 above)
Comment (markjhoy) [reply docs/reference/inference/service-amazon-bedrock.asciidoc:129]: Ah thanks! I could not figure out for the life of me where that error was coming from!

**Review by leemthompo — APPROVED** [review]
LGTM from writing perspective!

File: docs/reference/inference/service-amazon-bedrock.asciidoc (hunk 8 above)
Comment (markjhoy) [reply docs/reference/inference/service-amazon-bedrock.asciidoc]: FYI - you're correct here... theoretically, you _can_ use all three, but you shouldn't use `temperature` and `top_p` at the same time. For reference, see the [parameters in Amazon's Anthropic docs](https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-anthropic-claude-text-completion.html)

**Pull request merged**